### PR TITLE
Generate JSON Schema using shacl2code

### PR DESCRIPTION
**Pull request by JPEWdev**
Generates the JSON schema for validating JSON-LD documents using `shacl2code`

### CODE REVIEW

**Comment by JPEWdev** [comment]
@licquia  Here is the actions to generate the JSON schema. Note that the schema is still incorrect until the model & spec parser are fixed

**Review by goneall — APPROVED** [review]
Thanks @JPEWdev - This will be quite helpful for validation

File: .github/workflows/publish_v3.yml [hunk 1]
```diff
@@ -9,6 +9,8 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    env:
+      VERSION: v3.0
```
Comment (maxhbr) [inline .github/workflows/publish_v3.yml:13]: maybe no discussion for this PR: we will have patch releases? Shouldn't that be v3.0.0? Especially in the persistent URL for the context.

File: shacl2code-annotations.ttl
Comment (maxhbr) [inline shacl2code-annotations.ttl:1]: top level seems to be a rather prominent place for such a technical file. Can we move it somewhere else?

File: .github/workflows/publish_v3.yml [hunk 2]
```diff
@@ -32,12 +34,20 @@ jobs:
...
+            --input spdx-spec/shacl2code-annotations.ttl \
+            --context-url spdx-spec/docs/model/context.jsonld https://spdx.github.io/spdx-spec/${{ env.VERSION }}/model/context.jsonld \
+            jsonschema \
+            --output spdx-spec/docs/model/schema.json
       - name: Set git identity
         run: git config user.name ci-bot; git config user.email ci-bot@spdx.dev
         working-directory: spdx-spec
       - name: Sync gh-pages
-        run: git checkout gh-pages && git pull && git checkout development/v3.0
+        run: git checkout gh-pages && git pull && git checkout development/${{ env.VERSION }}
```
Comment (maxhbr) [inline .github/workflows/publish_v3.yml:49]: This is messy code and a cleaner way of doing that might be
```suggestion
        run: |
          git fetch origin
          git branch -f gh-pages origin/gh-pages
```

File: .github/workflows/publish_v3.yml [hunk 3]
```diff
@@ -32,12 +34,20 @@ jobs:
...
         working-directory: spdx-spec
       - name: Build docs
-        run: mike deploy v3.0 v3-draft -b gh-pages -p
+        run: mike deploy ${{ env.VERSION }} v3-draft -b gh-pages -p
```
Comment (maxhbr) [inline .github/workflows/publish_v3.yml:52]: `VERSION` is now a variable, but `v3-draft` is still hardcoded

File: .github/workflows/publish_v3.yml (hunk 1 above)
Comment (maxhbr) [reply .github/workflows/publish_v3.yml:13]: `v3.0` is still hard coded in multiple places, e.g. line 18

**Review by maxhbr — CHANGES_REQUESTED** [review]

File: .github/workflows/publish_v3.yml (hunk 2 above)
Comment (maxhbr) [reply .github/workflows/publish_v3.yml:49]: as `development/v3.0` is hardcoded in line 18, making it dependent on the variable here causes inconsistencies

Comment (JPEWdev) [reply .github/workflows/publish_v3.yml]: Reverted to hardcoded since it's inconsistent

File: shacl2code-annotations.ttl
Comment (JPEWdev) [reply shacl2code-annotations.ttl:1]: Done

File: .github/workflows/publish_v3.yml [hunk 4]
```diff
@@ -32,6 +32,16 @@ jobs:
...
       - name: Build model
         run: python3 spec-parser/main.py spdx-3-model/model spdx-spec/docs/model
+      - name: Copy JSON annotations
+        run: cp spdx-spec/serialization/jsonld/annotations.ttl spdx-spec/docs/model/jsonld-annotations.ttl
+      - name: Generate JSON schema
+        run: |
+          shacl2code generate \
+            --input spdx-spec/docs/model/ontology.rdf.ttl \
+            --input spdx-spec/docs/model/jsonld-annotations.ttl \
+            --context-url spdx-spec/docs/model/context.jsonld https://spdx.org/rdf/3.0.0/spdx-context.jsonld \
```
Comment (maxhbr) [inline .github/workflows/publish_v3.yml:42]: I am confused? why is there an argument before the url?
```suggestion
            --context-url https://spdx.org/rdf/3.0.0/spdx-context.jsonld \
```
Maybe I am just not aware of the API of your tool

**Review by maxhbr — APPROVED** [review]
One question, but looks fine

File: .github/workflows/publish_v3.yml (hunk 4 above)
Comment (JPEWdev) [reply .github/workflows/publish_v3.yml:42]: The tool lets you either specify a URL (with --context) or a local file and URL (with --context-url) in which case the data is read from the local file, but the URL is used in the bindings. The later is useful to generate bindings before the URL is published

**Comment by goneall** [comment]
@licquia @JPEWdev - Is this ready to be merged?

**Review by licquia — APPROVED** [review]

**Pull request merged**